secure_message_v2/application.py

```diff
@@ -24,14 +33,30 @@ def create_app():
     app.register_blueprint(messages_bp, url_prefix="/messages")
     app.register_blueprint(threads_bp, url_prefix="/threads")
 
+    @app.errorhandler(UAAError)
+    @app.errorhandler(JWTValidationError)
+    def handle_exception(e):
+        logger.error(e.message)
+        response = jsonify({"error": "Unauthorized"})
+        response.status_code = 401
+        return response
+
+    @app.before_request
+    def before_request():
+        if app.config["UAA_CHECK_ENABLED"]:
+            try:
+                get_uaa_token(app)
```

Comment:
This probably should have gone in another story, I have implemented the check, but haven't assigned it as there is nothing which then uses the token

Reply:
Having reviewed the flow I agree. I wrongly assumed UAA was involved in the token signing verification of the internal users. However, that token verification is done directly by the secure message service itself. We are likely to still need to make requests to UAA to retrieve details of internal users so happy to leave this lose end here.